Aetherium/Artifacts/ArtifactOfRegression.cs

```diff
@@ -141,36 +148,44 @@ public override void Hooks()
 
         private void RegressAIToLowerForm(On.RoR2.CharacterAI.BaseAI.orig_OnBodyDeath orig, RoR2.CharacterAI.BaseAI self, CharacterBody characterBody)
         {
-            if (NetworkServer.active && ArtifactEnabled)
+            CharacterMaster master = self.master;
+            if (ArtifactEnabled && NetworkServer.active && IsExisting(master) && IsAnEnemy(master))
             {
-                if (self.master && self.master.IsDeadAndOutOfLivesServer() && (self.master.teamIndex == TeamIndex.Monster || self.master.teamIndex == TeamIndex.Lunar))
+                var masterName = master.name.Replace("(Clone)", "");
+                if (RegressionLookup.ContainsKey(masterName))
                 {
-                    var masterName = self.name.Replace("(Clone)", "");
-                    if (RegressionLookup.ContainsKey(masterName))
+                    foreach (var children in RegressionLookup[masterName].children)
                     {
-                        var regressionMasterName = RegressionLookup[masterName].Item1;
-                        var regressionSpawnAmount = RegressionLookup[masterName].Item2;
-
-                        var masterPrefab = Resources.Load<GameObject>($"Prefabs/CharacterMasters/{regressionMasterName}");
-
-                        if (masterPrefab)
+                        GameObject childMasterPrefab;
+                        string childName = children.Key;
+                        RegressData.ChildData childData = children.Value;
+                        if (childData.resource)
                         {
-                            for(int i = 0; i < regressionSpawnAmount; i++)
+                            childMasterPrefab = childData.resource;
+                        }
+                        else
+                        {
+                            childMasterPrefab = Resources.Load<GameObject>($"Prefabs/CharacterMasters/{childName}");
+                        }
+                        if (childMasterPrefab)
+                        {
+                            for (int i = 0; i < childData.count; i++)
                             {
-                                var theta = (Math.PI * 2) / regressionSpawnAmount;
+                                var theta = (Math.PI * 2) / childData.count;
```

Comment:
for multiple child types this creates a seperate circle for each of them which means worst case they get spawned inside of each other, might want to pull the circle iterator out of the loop and divide by children.Sum(c -> c.count) instead

Reply:
Okay. I see what you mean.

Reply:
I see what you mean, but I'm pepega at Math. I'll require assistance when I work on this again.

Reply:
Please review my implementation. Also changed `var theta` to `float` since it becomes a `double` which is slow.